batchflow/opensets/pascal.py

```diff
@@ -0,0 +1,161 @@
+        """ Return the path in the archive to the mask which is .png image by its name"""
+        return os.path.join(dirname(self.SETS_PATH), 'SegmentationClass', name + '.png')
+
+    def download(self, path):
+        self.download_archive(path)
+        with tarfile.open(self.localname, "r") as archive:
+            train_ids = self._get_ids(archive, 'train')
+            test_ids = self._get_ids(archive, 'val')
+
+            train_images = np.array([self._extract(archive, self._imagepath(name)) \
+                                     for name in train_ids], dtype=object)
+            test_images = np.array([self._extract(archive, self._imagepath(name)) \
+                                    for name in test_ids], dtype=object)
+
+            train_masks = np.array([self._extract(archive, self._maskpath(name)) \
+                                    for name in train_ids], dtype=object)
+            test_masks = np.array([self._extract(archive, self._maskpath(name)) \
+                                   for name in test_ids], dtype=object)
+
+            self._train_index = DatasetIndex(np.arange(len(train_images)))
+            self._test_index = DatasetIndex(np.arange(len(test_images)))
+
+            return (train_images, train_masks), (test_images, test_masks)
+
+
+class PascalClassification(BasePascal):
+    """ Contains 11540 images and corresponding classes
+
+    Notes
+    -----
+    - Labels are represented by the one-hot vector of size 20. '1' stands for the presence of at least one object from
+    coresponding class on the image. '-1' stands for the absence. '0' indicates that the object is presented,
+    but can hardly be detected.
+    """
+    task = 'Main'
+    classes = ['aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus', 'car', 'cat', 'chair', 'cow',
+               'diningtable', 'dog', 'horse', 'motorbike', 'person', 'pottedplant', 'sheep', 'sofa', 'train',
+               'tvmonitor']
+
+    def download(self, path):
+        self.download_archive()
+        with tarfile.open(self.localname, "r") as archive:
+            d = defaultdict(list)
+            class_files = [os.path.join(self.SETS_PATH, self.task, name) + '_trainval.txt' for name in self.classes]
+            for class_file in class_files:
+                data = archive.extractfile(class_file).read()
+                for row in data.decode().split('\n')[:-1]:
+                    key = row.split()[0]
+                    value = int(row.split()[1])
+                    d[key].append(value)
+
+            train_ids = self._get_ids(archive, 'train')
+            test_ids = self._get_ids(archive, 'val')
+
+            train_images = np.array([self._extract(archive, self._imagepath(name)) for name in train_ids], dtype=object)
+            test_images = np.array([self._extract(archive, self._imagepath(name)) for name in test_ids], dtype=object)
+
+            train_labels = np.array([d[self._name(name)] for name in train_ids])
```

Comment:
Many model would gone mad with these labels. Shouldn't we transform them into more sensible format?